cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -663,8 +671,16 @@ private ApplicationWithPrograms deployApp(NamespaceId namespaceId, @Nullable Str
                                             ArtifactDetail artifactDetail,
                                             @Nullable KerberosPrincipalId ownerPrincipal,
                                             boolean updateSchedules) throws Exception {
-    // Enforce that the current principal has write access to the namespace the app is being deployed to
-    authorizationEnforcer.enforce(namespaceId, authenticationContext.getPrincipal(), Action.WRITE);
+    KerberosPrincipalId principalId = ownerPrincipal == null ? null : ownerPrincipal;
```

Comment:
Can we reuse `ownerPrincipal` name for namespace owner too? It makes reading the code easy.